### DISCUSSION TITLE
Plans for Sympa Release 6.2.72

### DISCUSSION THREAD

**Discussion by racke** [sympa-community/sympa#1633] [Release works]
Due to the delay releasing 6.2.70 we got already a lot of good stuff for 6.2.72. So I propose one beta release in about 4 weeks and the final release early January 2023.

**Comment by ikedas** [comment]
Agreed.:smile:

**Comment by ikedas** [comment]
I think these qualify for the 6.2.72 label:
- https://github.com/sympa-community/sympa/issues/872
- https://github.com/sympa-community/sympa/pull/1516
-  https://github.com/sympa-community/sympa/pull/1517
-  https://github.com/sympa-community/sympa/pull/1518

Once these fixes are merged, I'll switch the default branch to `main`.

**Comment by ikedas** [comment]
These may be merged before the next beta:
  - sympa-community/sympa#1533 Allow hyphens in command line option
  - sympa-community/sympa#1536  Bug on constom attribute datasource synchronization.

Below may or may not make it to the next beta. If this is not merged by Wednesday, please ignore it and work on a release. 
  - sympa-community/sympa#1538 RFC 8058 One-Click Unsubscribe

**Comment by racke** [comment]
I added #1533 to the 6.2.72 milestone.

**Comment by ikedas** [comment]
I added these to 6.2.72 and merged them:
- RFC 8058 One Click Unsubscribe [\#1210](https://github.com/sympa-community/sympa/issues/1210)
- SympaSOAP: List description with SOAP [\#1542](https://github.com/sympa-community/sympa/issues/1542)
- Some bugs with DKIM / ARC [\#1543](https://github.com/sympa-community/sympa/pull/1543)
- Custom attributes synchronization fail [\#1535](https://github.com/sympa-community/sympa/issues/1535)

No more changes are waiting.

**Comment by racke** [comment]
Beta will be released on Thursday.

**Comment by racke** [comment]
Done.

**Comment by racke** [comment]
The release announcement script lives currently at https://github.com/racke/sympa/tree/pr/github-release-script

**Comment by ikedas** [comment]
Thanks for release @racke!

**Comment by ikedas** [comment]
> The release announcement script lives currently at https://github.com/racke/sympa/tree/pr/github-release-script

The autolink and SMTP features are very good. But try-catch requires Perl 5.34 or later (along with feature pragma).

**Comment by ikedas** [comment]
What I noticed more.
  * Text::Markdown looks buggy. It outputs such as:
    ``` html
    <li><p>WWSympa: <code>msg</code> (ex. <code>arcsearch_id</code>) crashes by @ikedas in <a href="https://github.com/sympa-community/sympa/pull/1551">https://github.com/sympa-community/sympa/pull/1551</a></p>

    <h3>Other changes</h3></li>
    <li><p>Adding GH workflow to submit the PR for translation by @ikedas in <a href="https://github.com/sympa-community/sympa/pull/1515">https://github.com/sympa-community/sympa/pull/1515</a></p></li>
    ```
    A better alternative would be [CommonMark](https://metacpan.org/pod/CommonMark).

  * Newlines in text in the responses by REST API seem to be CR+LF. In our case this is not a problem because Sympa normalizes newlines during DKIM signing.  If DKIM signing is disabled, CR+LF and LF will be mixed.

**Comment by racke** [comment]
It is time to decide which changes should be added to 6.2.72.

**Comment by ikedas** [comment]
I think they'd be better to be added.
  * [The changes currently have "on going" label](https://github.com/sympa-community/sympa/labels/ready).
  * sympa-community/sympa#1565
  * sympa-community/sympa#1564, If the author @racke has no particular concerns.

**Comment by racke** [comment]
I added #1565 (straightforward change). #1564 definitely needs a review. I will check the "on going" ones. This label is confusing though. I would expert something like "ready to merge".

**Comment by ikedas** [comment]
The "on going" label is just a habit we have been using, so you can do what you think is better :smiley: .

**Comment by racke** [comment]
It sounds to me that there is still something happening in these PR's. so will change this label if you don't mind.

**Comment by ikedas** [comment]
I have no objection.

**Comment by racke** [comment]
Changed to "ready".

**Comment by ikedas** [comment]
[CVE-2023-24038](/advisories/GHSA-p983-rj5f-9qwf) (sympa-community/sympa#1573) has a relatively high CVSS score of 7.5.  I have already reported it to upstream (clintongormley/perl-html-stripscripts#3) and submitted a PR for the fix, but there is no indication that it will be merged.

Since sympa-community/sympa#1575 contains a monkey patch equivalent to the fix above, I would like to merge this into Sympa for now.

**Comment by racke** [comment]
Sounds good to me :+1: Will do the release on the weekend.

**Comment by ikedas** [comment]
Done.
Please freely merge the remaining PRs.

**Comment by racke** [comment]
Postponed due to positive private reasons. Trying to get it done today.

**Reply by racke** [reply]
@ikedas What is the workflow to merge the translations (#1554) into main?

**Reply by ikedas** [reply]
That PR is automatically pushed as a result of periodically (daily) running [support/sync_translation.sh](https://github.com/sympa-community/sympa/blob/main/support/sync_translation.sh), and it always updates the HEAD to the latest translation. So you can merge it at any time.

**Reply by ikedas** [reply]
also you can do it any time*s*: PR will be resubmitted by the bot when the translations are updated.

**Comment by ikedas** [comment]
@racke, are there any other problems?

On a personal note, I'd be very grateful if the new release would be available in April.

**Comment by ikedas** [comment]
Fedora changed the upstream of the `fontawesome-font` package  from Font Awesome 4 to 6: https://src.fedoraproject.org/rpms/sympa/pull-request/2
`sympa` packages on Fedora/EPEL depend on it.

If possible, I'd like to move to Font Awesome 6 (See #1656) at the time of 6.2.72.  @racke, when do you plan to release?

**Comment by racke** [comment]
How long would it take to prepare the move to Font Awesome 6 (see also my comments on https://github.com/sympa-community/sympa/issues/1656)?

**Reply by ikedas** [reply]
I think moving to FA6 can be done in a week.

**Comment by racke** [comment]
Sounds good to me. Please let me know when you have enough code to test it.

**Comment by ikedas** [comment]
@xavierba , packagers affected by this issue are Fedora, Debian and Ubuntu as far as I know. I'd like to hear your opinion if you have one.

**Reply by xavierba** [reply]
I think I can do with either FA4 or FA6, both are available in Fedora Rawhide. If needed, I can fallback to whichever version will be bundled with Sympa for earlier Fedora releases and EPEL.

**Comment by ikedas** [comment]
@xavierba , thanks for comment.

The transition to FA6 seems to be going well.  In addition, it turned out that we can also retire the Foundation Icons (we have downloaded tens of kilobytes for only 5 icons).

**Comment by ikedas** [comment]
PR #1657 has been finalized.  @racke, please feel free to merge it whenever you prefer.

**Comment by ikedas** [comment]
Hi @racke, is it likely that a release schedule will be set?

I have incorporated some of the features of 6.2.72 into our sympa.community server, and so far there seems to be no problems.

**Comment by ikedas** [comment]
Is it likely that a release schedule will be set?

**Comment by racke** [comment]
6.2.72 will be released on Thursday. Sorry for the delay.

**Comment by ikedas** [comment]
Thank you!

**Comment by racke** [comment]
Release done.

**Comment by ikedas** [comment]
@xavierba , please check #1681.

The problem is that, Sympa 6.2.72 requires [Mail-DKIM 0.57 or later](https://github.com/sympa-community/sympa/blob/6.2.72/cpanfile#L277) but EL7 has only 0.39 .

Should I submit  bug?

**Comment by ikedas** [comment]
@xavierba , one workaround is adding `perl-Mail-AuthenticationResults` package as a dependency of `sympa`.  It will avoid crash, though ARC support which is not provided by Mail-DKIM 0.39 won't be available.

**Comment by xavierba** [comment]
Le 2023-06-12 11:28, IKEDA Soji a écrit :
> @xavierba [1] , please check #1681 [2].
> 
> The problem is that, Sympa 6.2.72 requires Mail-DKIM 0.57 or later [3]
> but EL7 has only 0.39 .
> 
You may submit a bug to request that perl-Mail-DKIM be updated to a more 
recent version, but as this shipped with base RHEL, I don't think this 
will be granted, especially as RHEL 7 is getting really old now and will 
be EOL in a few months.

However, from #1681, I understand installing 
perl-Mail-AuthenticationResults fixed the issue, so I may add the 
dependency to sympa package.
Is that correct ?

**Reply by ikedas** [reply]
Thank you for rapid response.  I am in favor of adding that dependency to sympa.

**Reply by ikedas** [reply]
@xavierba , on the other issue #1685 I proposed a workaround.

**Comment by xavierba** [comment]
Is the patch intended to be applied for all releases or only EL7 ? I 
guess the latter, but I'd rather ask

**Reply by ikedas** [reply]
It should apply only to EL7, as it removes features that Mail-DKIM of EL7 does not support.

**Comment by xavierba** [comment]
That was my understanding, thanks for confirmation.
I'll update the bug report when I have an EL7 build with the 2 fixes.
